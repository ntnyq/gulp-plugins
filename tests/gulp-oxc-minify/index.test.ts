import { describe, it } from 'vitest'
import gulpOxcMinify, { oxcMinify } from '../../packages/gulp-oxc-minify/src'
import {
  createFakeFileCreator,
  createFile,
  testTransformFile,
  testTransformStream,
} from '../utils'
import type { Options } from '../../packages/gulp-oxc-minify/src'
import type { StreamCreator } from '../utils'

const createFakeFile = createFakeFileCreator(
  'tests/gulp-oxc-minify/fixtures/index.ts',
)

function runTests(streamCreator: StreamCreator<Options>) {
  describe('file', () => {
    it('Should ignore empty file', () =>
      testTransformFile({
        streamCreator,
        file: createFile(),
      }))

    it('Should minify files as expected', () =>
      testTransformFile({
        streamCreator,
        file: createFakeFile(),
      }))

    it('Should works well when option verbose set', () =>
      testTransformFile({
        streamCreator,
        file: createFakeFile(),
        pluginOptions: {
          verbose: true,
        },
      }))
  })

  describe('stream', () => {
    it('Should minify files', () =>
      testTransformStream({
        streamCreator,
        file: createFakeFile(),
      }))
  })
}

describe('named export', () => {
  runTests(oxcMinify)
})

describe('default export', () => {
  runTests(gulpOxcMinify)
})
