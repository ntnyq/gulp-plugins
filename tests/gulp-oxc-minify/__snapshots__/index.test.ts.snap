// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`default export > file > Should minify files as expected 1`] = `"const person={name:\`Alice\`,age:30};type PersonType=typeof person;type Animal={name:string;};(({name:\`Buddy\`,breed:\`Golden Retriever\`}) satisfies Animal);function identity<e extends number|string = string>(u:e): e{return u}identity(),identity(42);interface HasLength{length:number;}function logLength<e extends HasLength = string>(u:e): void{console.log(u.length)}logLength(\`Hello\`),logLength([1,2,3]);function add(e:number,u:number): number{return e+u}type AddFunctionType=typeof add;interface Box<e extends object = {message:string;}>{content:e;}type Task={title:string;description?:string;completed:boolean;};(({title:\`Learn TypeScript\`,completed:!1,priority:\`High\`}) satisfies Task);function wrapInArray<e = string>(u:e): e[]{return[u]}wrapInArray(),wrapInArray(42);"`;

exports[`default export > file > Should works well when option verbose set 1`] = `"const person={name:\`Alice\`,age:30};type PersonType=typeof person;type Animal={name:string;};(({name:\`Buddy\`,breed:\`Golden Retriever\`}) satisfies Animal);function identity<e extends number|string = string>(u:e): e{return u}identity(),identity(42);interface HasLength{length:number;}function logLength<e extends HasLength = string>(u:e): void{console.log(u.length)}logLength(\`Hello\`),logLength([1,2,3]);function add(e:number,u:number): number{return e+u}type AddFunctionType=typeof add;interface Box<e extends object = {message:string;}>{content:e;}type Task={title:string;description?:string;completed:boolean;};(({title:\`Learn TypeScript\`,completed:!1,priority:\`High\`}) satisfies Task);function wrapInArray<e = string>(u:e): e[]{return[u]}wrapInArray(),wrapInArray(42);"`;

exports[`default export > stream > Should minify files 1`] = `"const person={name:\`Alice\`,age:30};type PersonType=typeof person;type Animal={name:string;};(({name:\`Buddy\`,breed:\`Golden Retriever\`}) satisfies Animal);function identity<e extends number|string = string>(u:e): e{return u}identity(),identity(42);interface HasLength{length:number;}function logLength<e extends HasLength = string>(u:e): void{console.log(u.length)}logLength(\`Hello\`),logLength([1,2,3]);function add(e:number,u:number): number{return e+u}type AddFunctionType=typeof add;interface Box<e extends object = {message:string;}>{content:e;}type Task={title:string;description?:string;completed:boolean;};(({title:\`Learn TypeScript\`,completed:!1,priority:\`High\`}) satisfies Task);function wrapInArray<e = string>(u:e): e[]{return[u]}wrapInArray(),wrapInArray(42);"`;

exports[`named export > file > Should minify files as expected 1`] = `"const person={name:\`Alice\`,age:30};type PersonType=typeof person;type Animal={name:string;};(({name:\`Buddy\`,breed:\`Golden Retriever\`}) satisfies Animal);function identity<e extends number|string = string>(u:e): e{return u}identity(),identity(42);interface HasLength{length:number;}function logLength<e extends HasLength = string>(u:e): void{console.log(u.length)}logLength(\`Hello\`),logLength([1,2,3]);function add(e:number,u:number): number{return e+u}type AddFunctionType=typeof add;interface Box<e extends object = {message:string;}>{content:e;}type Task={title:string;description?:string;completed:boolean;};(({title:\`Learn TypeScript\`,completed:!1,priority:\`High\`}) satisfies Task);function wrapInArray<e = string>(u:e): e[]{return[u]}wrapInArray(),wrapInArray(42);"`;

exports[`named export > file > Should works well when option verbose set 1`] = `"const person={name:\`Alice\`,age:30};type PersonType=typeof person;type Animal={name:string;};(({name:\`Buddy\`,breed:\`Golden Retriever\`}) satisfies Animal);function identity<e extends number|string = string>(u:e): e{return u}identity(),identity(42);interface HasLength{length:number;}function logLength<e extends HasLength = string>(u:e): void{console.log(u.length)}logLength(\`Hello\`),logLength([1,2,3]);function add(e:number,u:number): number{return e+u}type AddFunctionType=typeof add;interface Box<e extends object = {message:string;}>{content:e;}type Task={title:string;description?:string;completed:boolean;};(({title:\`Learn TypeScript\`,completed:!1,priority:\`High\`}) satisfies Task);function wrapInArray<e = string>(u:e): e[]{return[u]}wrapInArray(),wrapInArray(42);"`;

exports[`named export > stream > Should minify files 1`] = `"const person={name:\`Alice\`,age:30};type PersonType=typeof person;type Animal={name:string;};(({name:\`Buddy\`,breed:\`Golden Retriever\`}) satisfies Animal);function identity<e extends number|string = string>(u:e): e{return u}identity(),identity(42);interface HasLength{length:number;}function logLength<e extends HasLength = string>(u:e): void{console.log(u.length)}logLength(\`Hello\`),logLength([1,2,3]);function add(e:number,u:number): number{return e+u}type AddFunctionType=typeof add;interface Box<e extends object = {message:string;}>{content:e;}type Task={title:string;description?:string;completed:boolean;};(({title:\`Learn TypeScript\`,completed:!1,priority:\`High\`}) satisfies Task);function wrapInArray<e = string>(u:e): e[]{return[u]}wrapInArray(),wrapInArray(42);"`;
